src/transmission_manager.rs

```diff
@@ -281,13 +285,14 @@ mod test {
             );
         }
         // TODO: figure out way to check math for n_attempts > 1
-        let expected_time_elapsed = f64::powf(params.generation_interval, n_attempts);
+        let expected_time_elapsed =
+            f64::powi(params.generation_interval, n_attempts.try_into().unwrap());
 
         assert!(((sum_end_times / f64::from(n_iter)) - expected_time_elapsed).abs() < 0.1);
     }
 
     #[test]
     fn test_variable_number_of_infection_attempts() {
```

Comment:
Is there a reason not to just have the function inlined?

Reply:
Thanks. I use the function multiple times to test multiple numbers of infection attempts now that I added in the proper math to get the expected time.